src/Table/index.jsx

```diff
@@ -1,16 +1,110 @@
 import React from 'react';
 import classNames from 'classnames';
+import FontAwesomeStyles from 'font-awesome/css/font-awesome.min.css';
+import isRequiredIf from 'react-proptype-conditional-require';
 import PropTypes from 'prop-types';
 
 import styles from './Table.scss';
+import Button from '../Button';
 
 class Table extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = props.tableSortable ?
+      {
+        sortedColumn: this.props.defaultSortedColumn,
+        sortDirection: this.props.defaultSortDirection,
+      }
+      :
+      {
+        sortedColumn: '',
+        sortDirection: '',
+      };
+
+    this.onSortClick = this.onSortClick.bind(this);
+  }
+
+  onSortClick(columnKey) {
+    let newDirection = 'desc';
+
+    if (this.state.sortedColumn === columnKey) {
+      newDirection = (this.state.sortDirection === 'desc' ? 'asc' : 'desc');
+
+      this.setState({
```

Comment:
There are two setStates here that have some redundnacy in the `sortDirection: newDirection` line. I could either leave them this way, so that there is only one setState call per sort click, or I can have a setState at line 38 that only sets `sortedColumn: columnKey`, and then another setState at line 43 that sets `sortDirection: newDirection`. There would be the potential for two sortState calls on one sortClick in that case. I believe setStates are batched together, so I'm not sure it would be a performance issue with regards to re-rendering, but just wanted to double-check here.

Reply:
I may be missing some blunt logic, but is there a downside to resetting the `sortColumn` to `columnKey` when it's the same column being toggled? So only have one `setState` that sets both values regardless of which column is being toggled?

Reply:
Oh, that's a very good point. I guess I was trying to focus too much on what's the absolute minimum change, but that change would simplify the code at a low cost. I guess if the page is going to re-render as a result of the set state anyway, it doesn't matter if I change only some or all of the state. Thanks!